packages/react-scripts/scripts/start.js

```diff
@@ -47,7 +47,11 @@ const openBrowser = require('react-dev-utils/openBrowser');
 const paths = require('../config/paths');
 const configFactory = require('../config/webpack.config');
 const createDevServerConfig = require('../config/webpackDevServer.config');
+const getClientEnvironment = require('../config/env');
+const react = require('react');
```

Comment:
We should probably require relative to the project root to make sure we're resolving the `react` the project source code would. We do this when resolving templates as well.

I think something like this would work:
`const react = require(require.resolve("react", { paths: paths.appPath }))`